lib/index.js

```diff
@@ -123,16 +123,25 @@ AxeBuilder.prototype.analyze = function(callback) {
     config = this._config,
     source = this._source;
 
-	inject(driver, source, config, function() {
-		driver
-			.executeAsyncScript(function(context, options, config) {
-				/*global document, axe */
-				if (config !== null) {
-					window.axe.configure(config);
-				}
-				window.axe.a11yCheck(context || document, options, arguments[arguments.length - 1]);
-			}, context, options, config)
-			.then(callback);
+	return new Promise(function(resolve, reject) {
+		inject(driver, source, config, function() {
+			driver
+				.executeAsyncScript(function(context, options, config) {
+					/*global document, axe */
+					if (config !== null) {
+						window.axe.configure(config);
+					}
+					window.axe.run(context || document, options || {})
+						.then(arguments[arguments.length - 1]);
+				}, context, options, config)
+				.then(function(results) {
+					if (callback) {
+						callback(results);
```

Comment:
I'm not sure if this actually works. The way `assert` / `expect` works is that it throws an error. If this callback throws an error, does it actually get caught by the test runner? I suspect it might get swallowed by the promise unless you put a `.catch` in every one of your tests. If it does, we may need to use a different promise library. We should at least add some tests to make sure. Something like:

```js
assert.throws(function () {
  new Builder({...}).analyze(function () { throw 'something broke'; });
});
```

Reply:
I can confirm your code sample does not throw an error.  But I assumed you would need a `.catch` to catch any errors–weren't they also swallowed before?

Reply:
@WilcoFiers pushed an update without the internal `catch`.